app/controllers/katello/api/v2/api_controller.rb

```diff
@@ -141,19 +141,19 @@ def skip_session
     protected
 
     def scoped_search_query(query, group)
-      if group && !query.to_sql.include?('GROUP BY')
+      if group && !query_has_group_by?(query)
         query.select(group).group(group)
       else
         query
       end
     end
 
     def scoped_search_total(query, group)
-      result = scoped_search_query(query, group).count
-      if result.instance_of?(Hash)
-        result.keys.length
+      count_query = scoped_search_query(query, group)
+      if query_has_group_by?(count_query)
+        count_query.length
       else
-        result
+        count_query.count
```

Comment:
After some hours I decided that using `.count` with `.group` just causes too many SQL errors

(example below)
```
PG::UndefinedFunction: ERROR:  function count(character varying, character varying) does not exist
LINE 1: SELECT COUNT(name, stream) AS count_name_stream, "katello_mo...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

Reverted back to using `.length`, but only when there's a GROUP BY clause. This seems to fix it.

Reply:
How about count_query.[size](https://medium.com/@craigsheen/count-vs-length-in-rails-4308e83f6292)?

Reply:
`size` does a SQL `SELECT COUNT` query if the relation isn't already loaded into memory. I want to avoid that because if the query also has a `group`, it results in the error above. Using `length` in that case guarantees that it won't do a SQL query.